### PR TITLE
Add errdefs/pkg package

### PR DESCRIPTION
**Pull request by dmcgowan**
Separate the error definitions from the packages which use them.

This will allow us to keep our error definitions in a 1.0 package that shouldn't ever need a 2.0 since we are basing the errors on a fixed set. 

The libraries which create and manipulate the errors are likely to change and possibly need major version bumps. If the error definitions are in the same go module, this makes the error types incompatible through type checking and requires use of the interfaces. While the interfaces may be better for version compatibility, `errors.Is` is the most idiomatic solution.

### CODE REVIEW

**Review by AkihiroSuda — APPROVED** [review]

File: pkg/go.mod [hunk 1]
```diff
@@ -0,0 +1,16 @@
+module github.com/containerd/errdefs/pkg
+
+go 1.23.1
```
Comment (cpuguy83) [inline pkg/go.mod:3]: Can this be max go1.22?
The other go.mod is 1.20.

**Review by cpuguy83 — CHANGES_REQUESTED** [review]

File: pkg/go.mod (hunk 1 above)
Comment (dmcgowan) [reply pkg/go.mod]: Updated

**Review by cpuguy83 — APPROVED** [review]

File: pkg/go.mod [hunk 2]
```diff
@@ -0,0 +1,16 @@
+module github.com/containerd/errdefs/pkg
+
+go 1.22
+
+require (
+	github.com/containerd/errdefs v0.2.0
```
Comment (samuelkarp) [inline pkg/go.mod:6]: I assume then we need to update this once we tag a new version?

Comment (dmcgowan) [reply pkg/go.mod:6]: Yeah, we should update once we have the split version for sure

**Review by samuelkarp — APPROVED** [review]

**Pull request merged**